custom_components/tuya_local/devices/alecoair_d16_dehumidifier.yaml

```diff
@@ -108,16 +108,34 @@ secondary_entities:
         unit: "%"
         class: measurement
   - entity: binary_sensor
-    name: Tank
     class: problem
-    category: diagnostic
+    name: Tank full
+    icon: "mdi:cup-water"
+    optional: true
+    dps:
+      - id: 19
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: true
+          - value: false
+  - entity: binary_sensor
+    class: problem
+    name: Fault
+    icon: "mdi:engine-off-outline"
```

Comment:
I don't think this icon is a substantial improvement over the generic problem icon for a fault indicator.  I think it is better to leave the icon at the default so that user themes can customise it when there isn't a strong case that it is an improvement.